vulnerability/cleaner.go

```diff
@@ -0,0 +1,51 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package vulnerability
+
+import (
+	"context"
+	"github.com/elastic/cloudbeat/resources/providers/awslib/ec2"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+type Cleaner struct {
+	log      *logp.Logger
+	provider snapshotCleaner
+}
+
+type snapshotCleaner interface {
+	DeleteSnapshot(ctx context.Context, ids ec2.EBSSnapshot) error
+}
+
+func NewVulnerabilityCleaner(log *logp.Logger, provider snapshotCleaner) Cleaner {
+	log.Debug("VulnerabilityCleaner: New")
+	return Cleaner{
+		log:      log,
+		provider: provider,
+	}
+}
+
+func (f Cleaner) CleanSnapshots(ctx context.Context, snapshots []ec2.EBSSnapshot) {
```

Comment:
Add unit tests. (Just make sure that the provider mock gets called over and over again)